src/main/java/net/quedex/api/user/OrderModificationSpec.java

```diff
@@ -13,27 +13,51 @@
     private final long clientOrderId;
     private final Integer newQuantity;
     private final BigDecimal newLimitPrice;
+    private final boolean postOnly;
 
-    public OrderModificationSpec(long clientOrderId, int newQuantity, BigDecimal newLimitPrice) {
-        checkArgument(newQuantity > 0, "newQuantity=%s <= 0", newQuantity);
-        checkArgument(newLimitPrice.compareTo(BigDecimal.ZERO) > 0, "limitPrice=%s <= 0", newLimitPrice);
-        this.clientOrderId = clientOrderId;
-        this.newQuantity = newQuantity;
-        this.newLimitPrice = newLimitPrice;
+    public OrderModificationSpec(final long clientOrderId,
+                                 final int newQuantity,
+                                 final BigDecimal newLimitPrice,
+                                 final boolean postOnly) {
+        this(clientOrderId, newQuantity, newLimitPrice, postOnly, null);
+    }
+
+    public OrderModificationSpec(final long clientOrderId, final int newQuantity, final BigDecimal newLimitPrice) {
+        this(clientOrderId, newQuantity, newLimitPrice, false, null);
     }
 
     public OrderModificationSpec(long clientOrderId, int newQuantity) {
-        checkArgument(newQuantity > 0, "newQuantity=%s <= 0", newQuantity);
-        this.clientOrderId = clientOrderId;
-        this.newQuantity = newQuantity;
-        this.newLimitPrice = null;
+        this(clientOrderId, newQuantity, null, false, null);
     }
 
-    public OrderModificationSpec(long clientOrderId, BigDecimal newLimitPrice) {
-        checkArgument(newLimitPrice.compareTo(BigDecimal.ZERO) > 0, "limitPrice=%s <= 0", newLimitPrice);
+    public OrderModificationSpec(final long clientOrderId, final BigDecimal newLimitPrice, final boolean postOnly) {
+        this(clientOrderId, null, newLimitPrice, postOnly, null);
+    }
+
+    public OrderModificationSpec(final long clientOrderId, final BigDecimal newLimitPrice) {
+        this(clientOrderId, null, newLimitPrice, false, null);
+    }
+
+    private OrderModificationSpec(final long clientOrderId,
+                                  final Integer newQuantity,
+                                  final BigDecimal newLimitPrice,
+                                  final boolean postOnly,
+                                  final Void dummy) {
```

Comment:
Why is this?

Reply:
I wanted to have single constructor that does all the checks, and other just forward parameters to it, applying defaults for missing fields. Most complete public constructor has signature:
```
public OrderModificationSpec(
    final long clientOrderId,
    final int newQuantity,
    final BigDecimal newLimitPrice,
    final boolean postOnly
);
```
Private constructor (the one receiving forwards) would have this signature:
```
public OrderModificationSpec(
    final long clientOrderId,
    final Integer newQuantity, // Integer, because it is optional
    final BigDecimal newLimitPrice,
    final boolean postOnly
);
```
But it turns out that you can't overload things on int / Integer, so I had to add one dummy parameter.